mlflow/store/artifact/artifact_repo.py

```diff
@@ -321,7 +325,12 @@ def download_trace_data(self) -> Dict[str, Any]:
             - `MlflowTraceDataNotFound`: The trace data is not found.
             - `MlflowTraceDataCorrupted`: The trace data is corrupted.
         """
-        raise NotImplementedError
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_file = Path(temp_dir, TRACE_DATA_FILE_NAME)
+            # temp_file always exists, so we raise exception if
```

Comment:
Can you elaborate on `temp_file always exists`?

Reply:
Ah it doesn't always exist, for azure blob and azure data lake, we use something like 
```
with open(local_path, "wb") as file:
            container_client.download_blob(remote_full_path).readinto(file)
```
In such case the temp_file always exists even if the remote_full_path doesn't exist.
I think I should raise MlflowTraceDataNotFound instead as even if spans are empty the uploaded trace_data shouldn't be None.

Reply:
Can we do the following?

```
        blob = container_client.download_blob(remote_full_path)
        with open(local_path, "wb") as file:
            blob.readinto(file)
```

Reply:
Had a try! You're right this raises error if the blob is not found